canlink/tracewriters/asciiwriter.go

```diff
@@ -0,0 +1,97 @@
+package tracewriters
+
+import (
+	"os"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"go.uber.org/zap"
+	"github.com/pkg/errors"
+)
+
+// Responsible for managing the ascii trace file
+type AsciiWriter struct {
+	traceFile *os.File
```

Comment:
The members will be common for any `Writer` class. Is there a way to abstract them out? I think we should have a generic `Writer` interface / struct (idk the Go term) and the Ascii and Json just provide the frame-to-text functionality.

Reply:
I was initially thinking this too, but I was having problems with JSON, because every trace file must start with `[` and end with `]`. With these requirements, I thought it was best for each writer to have their own implementations for creating and closing the trace file, this way in the json writers implementation for `CreateTraceFile` it can write a `[` initially, and in `CloseFile` it can write `]`. Essentially the JSON writer needed and initial trace file write, and a final trace file write to ensure the JSON is valid.

Reply:
https://jsonlines.org

If you use a .jsonl file then you don't need the opening and closing brackets. This format was specifically designed for log files.

There is a lot of similarity between the json and ascii classes that should be abstracted out, beyond just the structure. Ideally I think you should be able to write a single concrete "writer" class which takes a .frame_to_text function as an argument, since the only difference between and two writers is the format